Fixing groupped messages, again

sketchydroide opened the pull request:
fixes: https://github.com/Expensify/ReactNativeChat/issues/464

sketchydroide committed:
need to use the sorted list

sketchydroide committed:
updating the sorted actions on the update

tgolen reviewed (commented):
Changes look OK, but this could also be DRYed up by adding a method for `getSortedReportActions()`

sketchydroide committed:
Merge branch 'master' into afonseca_fix_group_sorted_sequence

sketchydroide committed:
centralising in a method to update and sort the report actions

sketchydroide committed:
Merge branch 'master' into afonseca_fix_group_sorted_sequence

sketchydroide committed:
the sorted action list is the value directly
also need to update on the render because the did update runs after the render.
